src/ErrorHandler.php

```diff
@@ -111,52 +100,53 @@ private function __construct(callable $callback, int $reservedMemorySize = self:
             // first call to the set_error_handler method would cause the PHP
             // bug https://bugs.php.net/63206 if the handler is not the first
             // one in the chain of handlers
-            set_error_handler([$this, 'handleError'], $this->capturedErrors);
-
-            $this->isRoot = true;
+            set_error_handler([$this, 'handleError'], E_ALL);
         }
 
         $this->previousExceptionHandler = set_exception_handler([$this, 'handleException']);
     }
 
     /**
-     * Registers this error handler and associates the given callback to the
-     * instance.
+     * Gets the current registered error handler; if none is present, it will register it.
+     * Subsequent calls will not change the reserved memory size.
      *
-     * @param callable $callback           callback that will be called when exception is caught
-     * @param int      $reservedMemorySize The amount of memory to reserve for the fatal error handler
+     * @param int|null $reservedMemorySize The requested amount of memory to reserve
      *
-     * @return self
-     *
-     * @throws \ReflectionException
+     * @return self The ErrorHandler singleton
      */
-    public static function register(callable $callback, int $reservedMemorySize = self::DEFAULT_RESERVED_MEMORY_SIZE): self
+    public static function registerOnce(int $reservedMemorySize = null): self
     {
-        return new self($callback, $reservedMemorySize);
+        if (null === self::$handlerInstance) {
+            self::$handlerInstance = new self($reservedMemorySize ?? self::DEFAULT_RESERVED_MEMORY_SIZE);
+        }
+
+        return self::$handlerInstance;
     }
 
     /**
-     * Sets the PHP error levels that will be captured by the Raven client when
-     * a PHP error occurs.
+     * Adds a listener to the current error handler to be called upon each invoked captured error;
+     * if no handler is registered, this method will instantiate and register it.
      *
-     * @param int  $levels  A bit field of E_* constants for captured errors
-     * @param bool $replace Whether to replace or amend the previous value
-     *
-     * @return int The previous value
+     * @param callable $listener A callable that will act as a listener; this callable will receive
+     *                           a single \ErrorException argument
      */
-    public function captureAt(int $levels, bool $replace = false): int
+    public static function addErrorListener(callable $listener): void
     {
-        $prev = $this->capturedErrors;
-
-        $this->capturedErrors = $levels;
-
-        if (!$replace) {
-            $this->capturedErrors |= $prev;
-        }
-
-        $this->reRegister($prev);
+        $handler = self::registerOnce();
+        $handler->errorListeners[] = $listener;
+    }
 
-        return $prev;
+    /**
+     * Adds a listener to the current error handler to be called upon each invoked captured exception;
```

Comment:
As a "soft" limit usually the descriptions of the methods goes to a new line after ~ 80 chars